go/pserver/service.go

```diff
@@ -51,21 +52,15 @@ type ParameterWithConfig struct {
 	Config []byte // parameter configuration in Proto Buffer format
 }
 
-// ParameterCheckpoint is Parameter and State checkpoint
-type ParameterCheckpoint struct {
-	ParamConfig ParameterWithConfig
-	State       []byte
-}
-
-// checkpoint signature
+// checkpointMeta saves checkpoint metadata
 type checkpointMeta struct {
 	UUID      string `json:"uuid"`
-	Md5sum    string `json:"md5sum"`
+	MD5       string `json:"md5"`
 	Timestamp string `json:"timestamp"`
 }
 
 // Checkpoint is the pserver shard persist in file
-type Checkpoint []ParameterCheckpoint
+type Checkpoint []parameterCheckpoint
```

Comment:
I am following @typhoonzero 's comment here for possible usage. 
https://github.com/PaddlePaddle/Paddle/pull/2716#discussion_r126294140
But it is true, single parameterCheckpoint is useless to the user. It should be use immediately out of pserver's package.